rs/consensus/src/consensus/validator.rs

```diff
@@ -595,22 +594,85 @@ fn get_notarized_parent(
         .map_err(|_| ValidationFailure::BlockNotFound(parent.clone(), height).into())
 }
 
-/// Collect the min of validated block proposal ranks in the range.
-fn get_min_validated_ranks(
+fn get_disqualified_ranks(
     pool: &PoolReader<'_>,
-    range: &HeightRange,
-) -> BTreeMap<Height, Option<Rank>> {
-    (range.min.get()..=range.max.get())
-        .map(|h| {
-            let height = Height::from(h);
-            (
-                height,
-                find_lowest_ranked_non_disqualified_proposals(pool, height)
-                    .first()
-                    .map(|block| block.rank()),
-            )
-        })
-        .collect()
+    membership: &Membership,
+    cfg: ReplicaConfig,
+    range: HeightRange,
+) -> RankMap {
+    let mut rank_map = RankMap::new(cfg);
+    let Some(proof_range) = pool.pool().validated().equivocation_proof().height_range() else {
+        return rank_map;
+    };
+
+    let narrowed_range = HeightRange::new(
+        std::cmp::max(range.min, proof_range.min),
+        std::cmp::min(range.max, proof_range.max),
+    );
+
+    for proof in pool
+        .pool()
+        .validated()
+        .equivocation_proof()
+        .get_by_height_range(narrowed_range)
```

Comment:
Would this be the same?
```suggestion
    for proof in pool
        .pool()
        .validated()
        .equivocation_proof()
        .get_by_height_range(range)
```

Reply:
done